packages/opentelemetry-core/test/utils/environment.test.ts

```diff
@@ -134,6 +136,14 @@ describe('environment', () => {
       assert.strictEqual(env.OTEL_EXPORTER_OTLP_TRACES_TIMEOUT, 12000);
     });
 
+    it('should parse OTEL_SDK_DISABLED despite casing', () => {
+      mockEnvironment({
+        OTEL_SDK_DISABLED: 'TrUe',
+      });
+      const env = getEnv();
+      assert.strictEqual(env.OTEL_SDK_DISABLED, true);
```

Comment:
Would you mind adding cases to cover negative values too?

Reply:
I added tests for insensitive "false" string and empty string. 

I'll create a separate PR for the warning mentioned in the specification for all the types. 

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#boolean-value